python/foglamp/common/configuration_manager.py

```diff
@@ -30,6 +31,56 @@
                               'URL', 'enumeration'])
 
 
+class CachedConfiguration(object):
+    """Cached Configuration Manager"""
+
+    MAX_CACHE_SIZE = 10
+
+    def __init__(self):
+        """
+        cache: value dictionary as per category_name
+        max_cache_size: Hold the 10 recently requested categories in the cache
+        hit: number of times an item is read from the cache
+        miss: number of times an item was not found in the cache and a read of the storage layer was required
+        """
+        self.cache = {}
+        self.max_cache_size = self.MAX_CACHE_SIZE
+        self.hit = 0
+        self.miss = 0
+
+    def __contains__(self, category_name):
+        """Returns True or False depending on whether or not the key is in the cache
+        and update the hit and data_accessed value only If True"""
+        if category_name in self.cache:
+            current_hit = self.cache[category_name]['hit']
+            self.cache[category_name].update({'date_accessed': datetime.datetime.now(), 'hit': current_hit + 1})
+            return True
+        return False
```

Comment:
update miss only if not in, handle `self.cache[category_name]['hit' | 'miss']` with try except keyError and assign current_hit|miss = 0 before update